match/dao/dao.go

```diff
@@ -112,10 +119,10 @@ func (dao *DAO) ListMatch() (*[]Match, error) {
 
 // CreateMatch creates a new match in the datastore, returning the newly created match
 func (dao *DAO) CreateMatch(input CreateMatchInput) (*Match, error) {
-	row := executeQueryWithRowResponse(dao.DB, "INSERT INTO match (userOne, userTwo, matchedOn) VALUES ($1, $2, NOW()) RETURNING *", input.UserOne, input.UserTwo)
+	row := executeQueryWithRowResponse(dao.DB, "INSERT INTO match (auth_id, userOne, userTwo, matchedOn) VALUES ($1, $2, NOW()) RETURNING *", input.UserOne, input.UserTwo)
```

Comment:
Just a thought here, since we're no longer distinguishing between user_ids and auth_ids, is there ever a case where userOne isn't going to be the same as auth_id?

Reply:
This is exactly the issue I flagged the other day, can see updates in the cancelled card on Monday.

In brief, there's two ways of modelling the Match service and we should support both.

This implementation:
```
Match: service {
  UserOne: User;
  UserTwo: User;
  ...
}
```

Alternative implementation:
```
Match: service {
  OtherUser: User;
  ...
}
```

`auth_id will become `created_by` in a future PR.

I'm happy to talk about this more offline, it's quite nuanced and I've got some good examples to illustrate the reason we're doing it this way.

Reply:
I just had a brief chat with @Lewiky and I think all is clear now

Reply:
For future reference:

```
User: service {
  firstName: string;
  lastName: string;
  #auth{by: email};
}
GuineaPig: service {
  name: String;
}
Pairing: service {
  // Implicit auth_id, soon to be changed to created_by
  GuineaPigOne: GuineaPig;
  GuineaPigTwo: GuineaPig;
}
```
here the created_by is always different to the guineaPigOne or Two ID